railties/test/commands/server_test.rb

```diff
@@ -44,6 +44,22 @@ def test_environment_with_rack_env
     end
   end
 
+  def test_caching_without_option
+    args = []
+    options = Rails::Server::Options.new.parse!(args)
+    assert_equal nil, options[:caching]
+  end
+
+  def test_caching_with_option
+    args = ["-C"]
```

Comment:
@kaspth I'm ambivalent about the whole thing as well, just added it as it seemed there was support in the previous PR. I'll go with the hivemind on it. I agree that `--toggle-caching` is superior though.